### PR TITLE
refactor(compiler_benchmark): simplify by using bootstrap and DI

### PR DESCRIPTION
**Pull request by tbosch**

### CODE REVIEW

File: modules/benchmarks/src/compiler/compiler_benchmark.ts [hunk 1]
```diff
@@ -1,82 +1,108 @@
...
-        print(`[${desc}] ...done, took ${elapsedMs} ms`);
-      };
-      PromiseWrapper.then(func(), onSuccess, null);
+function createBindings(multiplyTemplatesBy: number) {
+  return [
+    bind(ViewResolver)
+        .toFactory(() => new MultiplyViewResolver(
+                       multiplyTemplatesBy,
+                       [BenchmarkComponentNoBindings, BenchmarkComponentWithBindings])),
+    // Use DynamicChangeDetector as the view is too big for JitChangeDetector :-)
```
Comment (rkirov) [inline modules/benchmarks/src/compiler/compiler_benchmark.ts]: Do we fail reasonably in that case? Should we file an issue with Victor about this?

File: modules/benchmarks/src/compiler/compiler_benchmark.ts [hunk 2]
```diff
@@ -1,82 +1,108 @@
...
-import {
-  DomElementSchemaRegistry
-} from 'angular2/src/core/render/dom/schema/dom_element_schema_registry';
-
 export function main() {
   BrowserDomAdapter.makeCurrent();
   var count = getIntParameter('elements');
+  bootstrap(AppComponent, createBindings(count))
+      .then((ref) => {
+        var app = ref.hostComponent bindAction(
```
Comment (rkirov) [inline modules/benchmarks/src/compiler/compiler_benchmark.ts]: does this even run?

File: modules/benchmarks/src/compiler/compiler_benchmark.ts [hunk 3]
```diff
@@ -1,82 +1,108 @@
...
+
+function measureWrapper(func, desc) {
+  return function() {
+    var begin = DateWrapper.now();
+    print(`[${desc}] Begin...`);
+    var onSuccess = function(_) {
+      var elapsedMs = DateWrapper.toMillis(DateWrapper.now()) - DateWrapper.toMillis(begin);
+      print(`[${desc}] ...done, took ${elapsedMs} ms`);
     };
+    var onError = function(e) { DOM.logError(e); } PromiseWrapper.then(func(), onSuccess, onError);
```
Comment (rkirov) [inline modules/benchmarks/src/compiler/compiler_benchmark.ts]: missing new line.

File: modules/benchmarks/src/compiler/compiler_benchmark.ts (hunk 1 above)
Comment (tbosch) [inline modules/benchmarks/src/compiler/compiler_benchmark.ts]: Not sure. But the previous benchmark was also using DynamicChangeDetector, also that the numbers are comparable with Dart. Will change the comment.

File: modules/benchmarks/src/compiler/compiler_benchmark.ts [hunk 4]
```diff
@@ -1,82 +1,108 @@
...
+  };
+}
+
+
+class MultiplyViewResolver extends ViewResolver {
```
Comment (rkirov) [inline modules/benchmarks/src/compiler/compiler_benchmark.ts]: I think in TS all classes can be used as interfaces too, and that matches your usage here. So just `implements ViewResolver` should work.

Comment (tbosch) [inline modules/benchmarks/src/compiler/compiler_benchmark.ts]: I need to call `super`...

File: modules/benchmarks/src/compiler/compiler_benchmark.ts (hunk 2 above)
Comment (tbosch) [inline modules/benchmarks/src/compiler/compiler_benchmark.ts]: yes

File: modules/benchmarks/src/compiler/compiler_benchmark.ts (hunk 4 above)
Comment (rkirov) [inline modules/benchmarks/src/compiler/compiler_benchmark.ts]: Nevermind, I saw you have `super.resolve`, then why do you need to redeclare `_cache`?

**Comment by rkirov** [comment]
LGTM, definitely cleaner this way.

**Pull request closed without merging**